From 1876be41a4f47b529ed3e195f002ac8b11b2ca7e Mon Sep 17 00:00:00 2001
From: Giuseppe Maggio <jertlokk@gmail.com>
Date: Tue, 1 Jan 2019 16:09:16 +0000
Subject: [PATCH] angler: QCamera: shutdown threadName error

device/huawei/angler/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c:536:19: error: address of array 'poll_cb->threadName' will alway
s evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
    if (!poll_cb->threadName) {
        ~~~~~~~~~~^~~~~~~~~~
device/huawei/angler/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c:641:22: error: address of array 'cmd_thread->threadName' will al
ways evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
    if (!cmd_thread->threadName) {
        ~~~~~~~~~~~~~^~~~~~~~~~
2 errors generated.

Change-Id: I86823b961b5735214e53383bc1a1dcf002f3bd83
---
 .../QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c b/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c
index 0351dfd..c72dfd1 100644
--- a/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c
+++ b/camera/QCamera2/stack/mm-camera-interface/src/mm_camera_thread.c
@@ -533,7 +533,7 @@ int32_t mm_camera_poll_thread_launch(mm_camera_poll_thread_t * poll_cb,
     if(!poll_cb->status) {
         pthread_cond_wait(&poll_cb->cond_v, &poll_cb->mutex);
     }
-    if (!poll_cb->threadName) {
+    if (poll_cb->threadName[0] == '\0') {
         pthread_setname_np(poll_cb->pid, "CAM_poll");
     } else {
         pthread_setname_np(poll_cb->pid, poll_cb->threadName);
@@ -638,7 +638,7 @@ int32_t mm_camera_cmd_thread_launch(mm_camera_cmd_thread_t * cmd_thread,
                    mm_camera_cmd_thread,
                    (void *)cmd_thread);
 
-    if (!cmd_thread->threadName) {
+    if (cmd_thread->threadName[0] == '\0') {
         pthread_setname_np(cmd_thread->cmd_pid, "CAM_launch");
     } else {
         pthread_setname_np(cmd_thread->cmd_pid, cmd_thread->threadName);
